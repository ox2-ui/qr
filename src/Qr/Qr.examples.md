### Default

    <Qr
      value="My value"
    />

### Custom size

    <Qr
      size={200}
      value="My value"
    />
