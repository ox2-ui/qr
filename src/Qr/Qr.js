import React, { PropTypes } from 'react';
import qrcode from 'yaqrcode';

/**
 * Qr Component
 */
const Qr = ({ className, size, style, value, ...other }) => (
  <img
    className={className}
    src={qrcode(value, { size })}
    style={style}
    {...other}
  />
);

Qr.defaultProps = {
  size: 100,
};

Qr.propTypes = {
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * Gerated qr image size in pixels
   */
  size: PropTypes.number.isRequired,
  /**
   * Override the inline-styles of the image element.
   */
  style: PropTypes.object,
  /**
   * Value to generate qr from.
   */
  value: PropTypes.string.isRequired,
};

export default Qr;
