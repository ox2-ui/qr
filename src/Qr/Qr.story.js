import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Qr from './Qr';

storiesOf('Qr', module)
  .add('default', () => (
    <Qr
      value="My value"
    />
  ))
  .add('size 200', () => (
    <Qr
      size={200}
      value="My value"
    />
  ));
