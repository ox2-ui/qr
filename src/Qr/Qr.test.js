/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Qr from './Qr';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <Qr
      value="My value"
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders size 200 correctly', () => {
  const wrapper = mount(
    <Qr
      size={200}
      value="My value"
    />
  );

  expect(wrapper).toMatchSnapshot();
});
