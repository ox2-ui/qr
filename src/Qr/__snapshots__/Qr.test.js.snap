// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Qr
  size={100}
  value="My value"
>
  <img
    src="data:image/gif;base64,R0lGODdhZABkAIAAAAAAAP///ywAAAAAZABkAAAC/4SPqcHd5xiUKj0Ts6t89Y9pwbSFI3gBogZ+Z1utZJp29BrB3Ivn5v2y+VQUYilW7B15yuSwtns6j4upEjLrZZnB6NJarHKvGJKQK4Ya1Wcg8Iw1weWW4Va3BqO9cePcX0f1E4Yi9dWFQGPGR+dieKeTsYjEMylm6fU2iNg29bdHiUmpmceZ6cmoIjrpptc6psdKmJi02vi6hbsJCdh7Wdb42yu5qVvKSyoMi2NLyxIoSlYqAzwt0pw7m1xsKL1M7XsqOKytxf2sigd9i0oOSsoWqb7uLvvevq1uCgOP738Pap61JofUGCSojKDChcaiIaTHMOLDY/SyobMnMV+ngP+7IJZjFolYOEeH0nx0CJKfyHij2vGyZ/JiITupwMWMaZPdN2mfronD6dMZRoUtXrEUaRGpzqTgVs6CGeggx4E3USB7ujTqyWDnqrq4ui+f0npax431kNWZWJoDx/ZDpygtWK9Mzf5zWklux5FuJ0KleqtuzyZgG5Z1K7jmRLyG1Qbtw9Mj4b2Msbb9CLDyPjx4E1McNbSgZc6PzjWeEfr0vM6mY+1ASdouXNckz/6AHVI2688bj67dXBZo5tFdM8Xe7bloaXdSgTsueZjtxr99iT//nbP2v5e3o4ddXlejd87Cp3O9DF37OBvlf54ner49P4v0g1dUnJGt6P3Nt5L/DO9Xd4ANyN1ICeWnGzb2hVIcc5ZplouCzw22m22faTaVeQwOqNqF3XhjTkvLHHeXdZOhp6GFXnUYXnILqvgibW3U9997QcF4XYkfrieZfDjSaFSQzmG3I4nppfSWb/sVJuNeSVEnHWVFSqkbZhU2OSGVxlTX33IWstQMkTxmuUZof5zZ43gzuWFmKmgKZaNMIqKWFnxztibbX0/WGSCZdkKJCJI6ZsdkSoAKN5VUc8WJEphppsOXf1cG6pKTM7GCG5weJlopb5fqhOCBvT0mjpIIVhdmn/Ihmp+opZIqoqmOqteopFYCVKBzKd7qqp86oiXQpnTpJSyo2gA4nJzImB74ZS14JgnrspI1mxd60Cr7rK3YDpMqp9tCaCmtFnRb65jgeipuIuTGSe2WUd6pbnzEskhlsMZ2que9yqUnkFHrRjrimvjmFqG8ZFkbors7VTnwwacmm5p+yD36cJsOyjnbiTiKF+uDN2L48Zj5qicrf9Ki2+u+7e6oK34rZ6xhty1vSCieMvuXhr0XlzwxmTc/DHTQDxcAADs="
  />
</Qr>
`;

exports[`renders size 200 correctly 1`] = `
<Qr
  size={200}
  value="My value"
>
  <img
    src="data:image/gif;base64,R0lGODdhyADIAIAAAAAAAP///ywAAAAAyADIAAAC/4yPqcvtD6OctNr7gN68+w96lph1SoiaCYkx6QuT8QyPqsMeNHjebb8LqoTEjY0DyRmKxpXvh2ASl9LgsYl7BqpALDRa3VHDs6smqeU6kd81ubZ9x8wANBvM7J7beHlq7MWntDColnWno6UH17fHB7g3+CXp5piYV4IIWZdZVsn5GMdG+UN6GShaZIdquvmyGLopG9uaaukq1nlrC9pg9RnLOxtaWyi1uluc29grOKQIpazJ+3vICs38Ckz7PF19mvyn+4EcVn7+qc08DM7YvigdUm7oi/2eLr4ujI8Cy48Kzx4uTNZ2+cuW754xGgfbxeMxjp4LgfsQ9tu20JM+i/8A/xn0KARdwWbUEs7KOGcbx48rSQYkWG9aw4HyMKZZdq+lupjXZIL8NtEbzpM3Gap0OJRQ0Y46k84seVEfSnfsaKYc+fSlqnFaIUpdejUnUqNYjwZ9I3Jsza9CyYq1SlVpW6Zqp3D9CZbT1J1VoWrkGc5nU7dnA/8lOjds34cu75Z9DJjkXpP7Jkd9WzEynbSFG2/EDLRyXr6Z+2Z1rBl156ehZYymPJVzBMayWb++jLjn4dK8T0NerVqubsOKRSeOGzt4bbPCWYK+vVYhxc14cc+GDd25hFrUuVewvHte3q6eJ3in4FpwC/DFxR8nP2n69250MbCP6364ZMrUidH/137BfaTlR9x+1vX3SHr1BZgdfhGNB1J86tn3nx9ePfcec8l9tliDFsL1oYYQcsjbhhjqZ6GCKZIjYoYkmtihi3WlCKIftskYo3436lgdWn6tuGOBQRrI44se2vgjjS0WeaKQSzo5mI/BINjZfcGQdt6UWlI4YZLwcWNdlluOaR5FWPIXDXZdkslmmV2eeWCaYcrXZp2pjWQlmNGZZmefy7X0pX9zrtmmgE49ySKeI84ooZTBhfiblw/iKGkpSMoG6Z2MTsokXI2Ssd2RhDVH5IVVLuqpnI5GmilwgHIK5aaWyhHqcRKRGmhkMFa63qWwIkfnjIyJed2gnW55K7Gh/+XqjLGxjplssDW+iease5Yoba8w8alonK8Keu2uWkZLaJQADqiqqdiWm25IzBGYqK7ZwjtknbT1+Cyvp1KKbrcLqnGvuaUOPCyq+hZ7LXnaBCyswf3K2mS8tSaML4sMT3vsw6ka6a281GJycZK7anxwnuhV2/BFIbfWIL2IKosyxhCNHPO6FH+boJrHYlrhkCvXTDIdNLObYyQOA93yr1fqnC/PoxzdcdESk0qsyxwvWK/NBf4c9cGfTi3uoz2/bGbS/gIoNNRYI8qtm7tVrdy7cgv8tYrlTVwc3JH6hpnC80bY23yj6q2pq4VTXbPLhMs88NIf/1v2qGk/TTTizv/ezeDj50YenuA1LS6ywV//2XfFdUMT9t4Vt4rr3G5vLrnWmPPNOOtZWwt756kf7veKuTft+e8O7m74vr5zvbbqdLdOIum1+1p84CdrPrzocQtsO9vtwms3s6HHjjPejf8atOnMQ/x81+VfLfyy5KNudsqdh48w8Nhn677J8rcH+PQ7r46/Q+kvfTdDX/3GdzYHJZASA/ze/AzoNMht64Doyh6rEvc+8EWMdl5zIP8u2DXuHYp9CPRYRyx4uPWZ0C4g/B/9CKiuCMYweOYQG6W8B8OpyVCHNLzV7Ur3Qg8qMIUYXOAIN+i6jfXvdUOUGuh8dkTj2UWFjNvhE8mmQSn/uouKQrTi31rIQiYCrIj7w1wDRwfAyoWIi+4j4RXdGEXLbYWNcTzj9tKYvDIuLIBqRNrlWFY5NlWQj3lUohZ7F64vIgtNMOuj+pjmrub5CYfS453SLnm/G05yiZJU3grNCElAFrJQjCQk2jgpx60l5Y05i5OhstjFJJ6vbVW8HhCxuBUjahKMoIRlyKKHwlp2kojUI9kvZylEY67SlCXU4jGHCccJItJ+xJTgA2lZyWjm8pPEQ2bBHkm9cSlSmJbMZAGlJsIQHvB04LQmqGzozkjeEXdNzGYy/ehCPY7zluJTZjGPAc/2xVF87DynPYMpu3QalKDt8mc8fag9XVKs/5HbdCIzFfrBap6Sg9MEW/wYKtGMAvNvFA2jRR35UID6j5qedCYkKdnSe6qThFwaJblQdNEMXtObJGWmKFWJUoFKM6Cz4+kuy+i8g3JOqCkNZEJl2UZ+KjWcmPxkTfP1TYM+U6pZZepGI8rVOi61l/G0mlUzl7GxOlR4Se0qWWe4Qw5GFZtbpSs+4UpUjBJsmYP7KAXF6pNu/lSe5DypTVVqScHytaKx1OZXc7gcxfZ1goOFKTfv2hjJvg2xhZVdSQmLTHTqFKibnSxOgwra0wxtlHpN6uTSaque8mufHXVtD9VG2sciEZpoFak9X4tVv4J0t63cKSsN+dQgxlWWwP/lIW+b+0PkDle04PLtcUuGW+cStYHIy20zO2hOdSWHuy/NKyrtGF6PZoi8l1su/dCrz/baJLb0zSd1DQjfHK7WaPX1rlyzu1/t6je7hvUvVAGM4H9S1nqThNNpF3pbWNZ2k9DFJV6vmtrRFpXC7xUVazX8WS5S2MHevS5EQ7phDiP3lR9GcYjJqGIS73WmYhyo+fR64oi5FYpUpeklj3dhz6o1wWUF6yFlylgeNxW2WiXwj4HcXQFjt79v/e6OkWxSozZZvsolrl1VBuWxWnbFwjXyVD9HWzM318TlXSmwKNdijabYpU5NIEdvLJ06y7mgQa7wc/GcG926V1s79bP/VCfMljiPVM+d5fN3WztfRWtZ0IBug4zXDCEWB5fGjT6wgpMMOE1bGcaNLWdYafVkKgf4xe1kK4gx271K5zeVUq4rlmstDlErWbdzPbNXM/wqXVt4zqWO3n9RbWchcxm/Q/alh9/5Z2E79tF4dHV6K7tOWfvVrWudse/uLG0dMzi+Rb72Yk8Z7gJTO71ePB64nz3qZdN5yw8GEnOVncij2mvc98Wwd9KdXA5rlqV/JRTAuylIfgfc3/I5uMITru+F91bK6g6wivs9aWKPWdzy/i2ZBr7uGhs8zby9aYltvDw3T9fSzS5tvYF9WdSuHI0uJnKWIa3ybFe3ytg2N2rpWjhyRoe22vFeVcEhHG2Sn5qxogZ6oe8d5odXmkrFpjWiid5zkSNdzt0G+IBNy2uUcx3W8J43tA89dRpe/cgpx/q5id3rdtNbvAKMun1pDTqvn1fsF++73y9eAAA7"
  />
</Qr>
`;
